presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -80,6 +81,32 @@ public void testPrimitiveTypes()
             checkRepresentation(connectedStatement.getStatement(), "true", Types.BOOLEAN, true);
             checkRepresentation(connectedStatement.getStatement(), "'hello'", Types.VARCHAR, "hello");
             checkRepresentation(connectedStatement.getStatement(), "cast('foo' as char(5))", Types.CHAR, "foo  ");
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123'", Types.VARCHAR,
+                    (rs, column) -> assertEquals(rs.getLong(column), 123L));
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR ''", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getLong(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: ");
+            });
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123e-1'", Types.VARCHAR,
+                    (rs, column) -> assertEquals(rs.getLong(column), 12));
+
+            checkRepresentation(connectedStatement.getStatement(), "DOUBLE '123.456'", Types.DOUBLE,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 123.456D));
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123'", Types.VARCHAR,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 123D));
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR ''", Types.VARCHAR, (rs, column) -> {
```

Comment:
the `VARCHAR ''` case is tested above already, let's merge these two together

(you can have multiple assertions for in single `checkRepresentation` call, just use `{ ... }` lamnbda)

Reply:
fixed